### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by klamike**
Several mostly cosmetic changes.

- Changed conda environments to have the same name and explicitly grab signac from conda-forge (this is default with mamba, but not with conda)
- Changed the plmnist installation (no more holder directory & using pyproject.toml)
- Cleaned up the comments + code in project.py and init.py, refactored the seed analysis into src/seed_analysis.py
- Changed the phoenix environment to allow submission from a compute node
- Changed the final part post-condition to look at number of lines instead of just if the file exists

### CODE REVIEW

File: project/project.py [hunk 1]
```diff
@@ -1,606 +1,222 @@
...
+    # signac takes care of the writing - we just add attributes to job.document
+    # note that this file isn't used by the project - it's just here for demonstration.
+
+    # note that we can access the statepoint variables (from init.py) via job.statepoint
+    job.document.start_time = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
+    job.document.seed = job.statepoint.seed_int
+
+    # similarly, there is a project document that can also store some data.
+    project = get_project()
+    project.document.num_jobs = len(project)
```
Comment (bcrawford39GT) [inline project/project.py]: get rid of these lines, as they are not needed, as they are dynamically changing or could change

**Comment by bcrawford39GT** [comment]
Waiting to approve, but pyTorch seems to be having problems at the moment

**Comment by klamike** [comment]
Seems to be working now. Tested on Phoenix with mamba.

**Review by bcrawford39GT — APPROVED** [review]
LGTM.  Tested and approved

**Pull request merged**